src/main/java/io/camunda/testing/assertions/ProcessInstanceAssertions.java

```diff
@@ -351,21 +351,23 @@ public ProcessInstanceAssertions isWaitingExactlyAtElements(final String... elem
    * Verifies the expectation that the process instance is currently waiting to receive one or more
    * specified messages.
    *
-   * @param messageNames Names of the messages
+   * @param messageNamesVarArg Names of the messages
    * @return this {@link ProcessInstanceAssertions}
    */
-  public ProcessInstanceAssertions isWaitingForMessage(final String... messageNames) {
-    final List<String> notCreatedProcessMessageSubscriptions =
-        Arrays.stream(messageNames)
-            .filter(((Predicate<String>) this::isWaitingForMessage).negate())
-            .collect(Collectors.toList());
+  public ProcessInstanceAssertions isWaitingForMessage(final String... messageNamesVarArg) {
+    final List<String> messageNames = Arrays.asList(messageNamesVarArg);
+    final List<String> openMessageSubscriptions = getOpenMessageSubscriptions();
+
+    final List<String> differences = messageNames.stream()
+        .filter(element -> !openMessageSubscriptions.contains(element))
+        .collect(Collectors.toList());
 
-    assertThat(notCreatedProcessMessageSubscriptions.isEmpty())
+    assertThat(openMessageSubscriptions)
```

Comment:
:thought_balloon: If you fall back to the AssertJ default error message, you could save the lines for identifying the differences.